packages/hebao_v1/contracts/base/MetaTxModule.sol

```diff
@@ -88,21 +86,26 @@ contract MetaTxModule is BaseModule
     /// @param gasToken The token to pay the relayer as fees. Use address(0) for Ether.
     /// @param gasPrice The amount of `gasToken` to pay per gas. 0 is a valid value.
     /// @param gasLimit The max amount of gas that can be used by this meta transaction,
-    ///                excluding an extra `GAS_OVERHEAD` for paying the relayer fees.
+    ///                 excluding the `gasOverhead`.
+    /// @param gasOverhead The extra amount of `gasToken` paid for data storage and
+    ///                    extra fee-payment internal transaction.
     /// @param signatures Signatures.
     function executeMetaTx(
         bytes   calldata data,
         uint    nonce,
         address gasToken,
         uint    gasPrice,
         uint    gasLimit,
+        uint    gasOverhead,
         bytes[] calldata signatures
         )
         external
         payable
     {
+        require(gasLimit > 0, "INVALID_GAS_LIMIT");
+
         uint startGas = gasleft();
-        require(startGas >= gasLimit, "OUT_OF_GAS");
+        require(startGas >= gasLimit.add(gasOverhead), "OUT_OF_GAS");
```

Comment:
I would just do `require(startGas >= gasLimit, "OUT_OF_GAS");` because the `gasOverhead` gas is already used up before `gasleft()` is stored in `startGas`.